src/YoutubeAtom.tsx

```diff
@@ -90,78 +90,123 @@ export const youtubePlayerState = {
+        );
+    }
+
+    if (
+        sharedYTData[assetId].pastProgressPercentage < 75 &&
+        75 < percentPlayed
+    ) {
+        eventEmitters.forEach((eventEmitter) =>
+            eventEmitter('75' as VideoEventKey),
+        );
+    }
 
-// use booleans make sure that only 1 event has been sent per video
-const eventState: { [key: string]: boolean } = {
-    play: false,
-    end: false,
-    '25': false,
-    '50': false,
-    '75': false,
+    sharedYTData[assetId].pastProgressPercentage = percentPlayed;
+
+    // we recursively set set timeout as a way of only having one interval at a time querying
+    sharedYTData[assetId].progressTrackerTimoutId = window.setTimeout(
+        () => intervalProgressTracker({ eventEmitters, player, assetId }),
+        3000,
+    );
 };
 
 export const onPlayerStateChangeAnalytics = ({
     e,
-    setHasUserLaunchedPlay,
     eventEmitters,
     player,
+    assetId,
 }: {
     e: YoutubeStateChangeEventType;
-    setHasUserLaunchedPlay: (userLaunchEvent: boolean) => void;
     eventEmitters: ((event: VideoEventKey) => void)[];
     player: YoutubePlayerType;
+    assetId: string;
 }): void => {
     switch (e.data) {
         case youtubePlayerState.PLAYING: {
-            setHasUserLaunchedPlay(true);
-
-            if (!eventState['play']) {
+            if (!sharedYTData[assetId].hasPlayEventBeenFired) {
                 eventEmitters.forEach((eventEmitter) => eventEmitter('play'));
-                eventState['play'] = true;
+                sharedYTData[assetId].hasPlayEventBeenFired = true;
             }
 
             // Need to remove previous setInterval if already exists
-            if (progressTracker) clearInterval(progressTracker);
-
-            // NOTE: you will not be able to set React state in setInterval
-            // https://overreacted.io/making-setinterval-declarative-with-react-hooks/
-            progressTracker = setInterval(async () => {
-                const currentTime = await player.getCurrentTime();
-                const duration = await player.getDuration();
-
-                // Note that getDuration() will return 0 until the video's metadata is loaded
-                // which normally happens just after the video starts playing.
-                if (duration === 0) return;
-
-                const percentPlayed = Math.round(
-                    (currentTime / duration) * 100,
-                ) as number;
-
-                if (
-                    `${percentPlayed}` in eventState &&
-                    !eventState[percentPlayed]
-                ) {
-                    eventEmitters.forEach((eventEmitter) =>
-                        eventEmitter(`${percentPlayed}` as VideoEventKey),
-                    );
-                    eventState[percentPlayed] = true;
-                }
-            }, 500);
+            if (sharedYTData[assetId].progressTrackerTimoutId) {
+                clearTimeout(sharedYTData[assetId].progressTrackerTimoutId);
```

Comment:
I'm wondering if this is needed? Now we're not using setInterval there shouldn't be a need to clear anything. At the end of each check, we either decide that we're complete, or if not, we set the next timeout check.

If there's a situation where a user might click stop and then a check event fires. I think that in this situation we can check if the video is playing or not and use that to decide whether to set the next check.